src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/set_media_clock_timer_request.cc

```diff
@@ -151,7 +158,33 @@ bool SetMediaClockRequest::isDataValid() {
            (update_mode == mobile_apis::UpdateMode::COUNTDOWN)) ||
           ((end_time_in_seconds < start_time_in_seconds) &&
            (update_mode == mobile_apis::UpdateMode::COUNTUP))) {
-        SDL_LOG_INFO("Invalid data");
+        std::string update_mode_name;
+        smart_objects::EnumConversionHelper<
+            mobile_apis::UpdateMode::eType>::EnumToString(update_mode,
+                                                          &update_mode_name);
+        info = "Start time must be " +
+               std::string((end_time_in_seconds > start_time_in_seconds)
+                               ? "after"
+                               : "before") +
+               " the end time for update mode " + update_mode_name;
+        SDL_LOG_INFO("Invalid data: " << info);
+        return false;
+      }
+    }
+  }
+
+  std::vector<std::string> indicator_keys{strings::forward_seek_indicator,
+                                          strings::back_seek_indicator};
+  for (auto& key : indicator_keys) {
+    if (msg_params.keyExists(key)) {
+      mobile_apis::SeekIndicatorType::eType seek_indicator_type =
+          static_cast<mobile_apis::SeekIndicatorType::eType>(
+              msg_params[key][strings::type].asUInt());
+      if (seek_indicator_type == mobile_apis::SeekIndicatorType::TRACK &&
+          msg_params[key].keyExists(strings::seek_time)) {
+        info =
+            "The seekTime parameter is not applicable for indicator type TRACK";
```

Comment:
is it normal to reject messages for extraneous parameters?

Reply:
Not usually for extra params but we do reject for incompatible params in certain cases (For example https://github.com/smartdevicelink/sdl_core/blob/master/src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/button_press_request.cc#L111)

The seek indicator type `TRACK` is incompatible with the seekTime parameter since `TRACK` would skip directly to the next/previous track regardless of the value in the seekTime param. Having a seek duration in this case wouldn't really make sense.